Add downstream tests

d-montgomery opened the pull request:
To avoid future headaches I'm proposing we add some downstream tests for LMeX and PeleC. I think the following should catch any potential issues from a PelePhysics PR:

- [x] clang-tidy for LMeX (EB-OFF only)
- [x] clang-tidy for PeleC 
- [x] FlameSheet for LMeX
- [x] PMF for PeleC
- [x]  Edit `workflow_dispatch` to only trigger for `development` branch

Let me know if you have any thoughts/suggestions @jrood-nrel , @marchdf and @baperry2.

When we are ready to merge, we'll want to edit the `workflow_dispatch` to read:
~~~
on:
  workflow_dispatch:
  push:
    branches: [development]
  pull_request:
    branches: [development]
~~~

I had to set the push branches to all to demonstrate the tests work as expected in this PR.

d-montgomery committed:
Add clang-tidy test for LMeX

d-montgomery committed:
Remove 'on push'

d-montgomery committed:
switch to pull_request_target for now

d-montgomery committed:
test in fork

d-montgomery committed:
Fix build dir

d-montgomery committed:
Update clang-tidy

d-montgomery committed:
Simplify pointing to PP PR

d-montgomery committed:
Add PeleC clang-tidy test

d-montgomery committed:
Start downstream tests after PelePhysics-CI

d-montgomery committed:
add cancel-in-progress to downstream tests

d-montgomery committed:
Test concurrency

d-montgomery committed:
double check correct SHA for PP PR in LMeX and C

d-montgomery committed:
Try again

d-montgomery committed:
Add RegTests/FlameSheet from LMeX

d-montgomery committed:
Fix invalid identifier for test

d-montgomery committed:
Make LMeX-clang-tidy need FlameSheet test

d-montgomery committed:
Add PeleC-PMF test

d-montgomery committed:
Merge branch 'development' into downstream-tests

d-montgomery committed:
Remove extra line breaks

d-montgomery commented:
Looks like the changes I pulled in from #605 are causing the clang-tidy tests to fail.

baperry2 commented:
Show's why this PR is needed haha! Can you apply the fixes as part of this PR?

d-montgomery committed:
Fix clang-tidy error from #605

d-montgomery committed:
Formatting

baperry2 reviewed (approved):
Thanks for this. Looks good. But let's omit the PeleLMeX EB-ON clang-tidy just to keep the total number of tests down; I don't think it should provide anything the others don't (The PeleC build is EB-ON).

d-montgomery committed:
Remove EB-ON for LMeX

d-montgomery committed:
Edit workflow_dispatch for development only

d-montgomery commented:
I removed the PeleLMeX EB-ON clang-tidy test and changed the `workflow_dispatch` to trigger for pushes and pull requests for development only. This should be good to go now.

jrood-nrel commented on .github/workflows/downstream.yml:
```diff
@@ -0,0 +1,323 @@
+
+          sudo apt-get install -y --no-install-recommends \
+              build-essential    \
+              gcc-10 \
```
Since GCC should already be available I don't think we need this.

jrood-nrel commented on .github/workflows/downstream.yml:
```diff
@@ -0,0 +1,323 @@
+
+    env:
+      BUILD_DIR: ${{ github.workspace }}/build-clang-tidy-lmex
+      use_eb: "OFF"
```
Doesn't look like this line is necessary.

jrood-nrel commented on .github/workflows/downstream.yml:
```diff
@@ -0,0 +1,323 @@
+    runs-on: ubuntu-24.04
+
+    env:
+      BUILD_DIR: ${{ github.workspace }}/build-clang-tidy-c
```
I think we should be consistent with `BUILD_DIR` and either use it in all the jobs, or don't use it in all the jobs. I would recommend not using it and saving two lines.

jrood-nrel commented on .github/workflows/downstream.yml:
```diff
@@ -0,0 +1,323 @@
+          mpirun -n 2 ./PeleLMeX2d.gnu.MPI.ex input.2d-regt amr.max_step=2 amr.plot_int=-1 amr.check_int=-1 amrex.abort_on_unused_inputs=1 amr.n_cell=32 64 
+
+  PeleC-PMF:
+    name: PeleC GNU@9.3 MPI Run [PMF]
```
I think it would be better to avoid listing GCC versions here.

jrood-nrel commented:
This is a good idea. Thanks for putting the work into this. I wish PeleC didn't require so much building.

baperry2 commented:
Hmm for the Clang-Tidy here we don't really need to build everything. We just want clang-tidy to hit all the relevant pieces of PelePhysics. A combination like `MassCons`, `Soot-Flame`, and `Spray-EB` would probably be sufficient. Is there an easy way we could set it up to tell CMake to build only a particular subset of cases during the clang-tidy test?

d-montgomery committed:
Simplifications per Jon

d-montgomery committed:
Missed BUILD_DIR in run commands

d-montgomery commented:
Thanks for the suggestions @jrood-nrel. They should all be incorporated now.

> Hmm for the Clang-Tidy here we don't really need to build everything. We just want clang-tidy to hit all the relevant pieces of PelePhysics. A combination like `MassCons`, `Soot-Flame`, and `Spray-EB` would probably be sufficient. Is there an easy way we could set it up to tell CMake to build only a particular subset of cases during the clang-tidy test?

This is maybe a naive solution, but we could add a flag in CMake for LMeX and C, something like `PELE_UPSTREAM`, similar to what we do for the regression tests [here](https://github.com/AMReX-Combustion/PeleLMeX/blob/f82f7fa75b496ea0d556dab5e466fa6133b05f1c/Tests/CMakeLists.txt#L229) that includes the exact subset of tests we want to run from PelePhysics.

jrood-nrel commented:
I think we already do this https://github.com/AMReX-Combustion/PeleC/blob/2a001f023ef07fffbf9590e9ca6240170b34713d/CMakeLists.txt#L29

d-montgomery commented:
Thanks for pointing this out @jrood-nrel. I already have `DPELE_EXCLUDE_BUILD_IN_CI:BOOL=ON` in the PeleC clang-tidy test.  We could do something similar in LMeX, but those tests already take a lot less time to build/run than the PeleC test.

baperry2 commented:
`PELE_EXCLUDE_BUILD_IN_CI` kind of does we what we want, but I think we could go much further here and exclude almost everything. @dmontgomeryNREL up to you if you want to add the `PELE_UPSTREAM` flag to PeleC/PeleLMeX or just be satisfied with what gets removed by `PELE_EXCLUDE_BUILD_IN_CI`

d-montgomery commented:
For now I think we should merge as is so we have some downstream testing setup. We can always refine in the future.
